File: Java Back End/src/InfoSecCooker/Communication/Messages/CommandMessages/GraphBuilding/SetBufferSizeOfTaskCommandMsg.java
package InfoSecCooker.Communication.Messages.CommandMessages.GraphBuilding;

public class SetBufferSizeOfTaskCommandMsg extends GraphBuildingCommandMsg
{
    Long nodeId;
    Integer bufferSize;

    public SetBufferSizeOfTaskCommandMsg(Long nodeId, Integer bufferSize)
    {
        this.nodeId = nodeId;
        this.bufferSize = bufferSize;
    }

    public Long getNodeId()
    {
        return nodeId;
    }

    public Integer getBufferSize()
    {
        return bufferSize;
    }
}

File: JavaScript Front End/src/Messages/CommandMessages/GraphBuilding/SetBufferSizeOfTaskCommandMsg.js
const GraphBuildingCommandMsg = require('./GraphBuildingCommandMsg.js').GraphBuildingCommandMsg;

class SetBufferSizeOfTaskCommandMsg extends GraphBuildingCommandMsg
{
    constructor(nodeId, bufferSize)
    {
        super();
        this.jsonMessage["@type"] = this.jsonMessage["@type"].concat("SetBufferSizeOfTaskCommandMsg");
        this.jsonMessage["nodeId"] = nodeId;
        this.jsonMessage["bufferSize"] = bufferSize;
    }
}

module.exports = {SetBufferSizeOfTaskCommandMsg};

